provider/spark.go

```diff
@@ -730,15 +746,63 @@ func (spark *SparkOfflineStore) RegisterResourceFromSourceTable(id ResourceID, s
 }
 
 func (spark *SparkOfflineStore) CreateTransformation(config TransformationConfig) error {
+	return spark.sqlTransformation(config, false)
+}
+
+func (spark *SparkOfflineStore) sqlTransformation(config TransformationConfig, isUpdate bool) error {
+	updatedQuery, sources, err := updateQuery(config.Query, config.SourceMapping)
+	if err != nil {
+		return err
+	}
+
+	transformationDestination := spark.Store.ResourcePath(config.TargetTableID)
+	exists, err := spark.Store.ResourceExists(config.TargetTableID)
+	if err != nil {
+		return err
+	}
+
+	if !isUpdate && exists {
+		return fmt.Errorf("transformation %v already exists at %s", config.TargetTableID, transformationDestination)
+	} else if isUpdate && !exists {
+		return fmt.Errorf("transformation %v doesn't exist at %s and you are trying to update", config.TargetTableID, transformationDestination)
+	}
+
+	sparkArgs := spark.Store.SparkSubmitArgs(transformationDestination, updatedQuery, sources, Transform)
+	if err := spark.Executor.RunSparkJob(sparkArgs); err != nil {
+		return fmt.Errorf("spark submit job for transformation %v failed to run: %v", config.TargetTableID, err)
+	}
 	return nil
 }
 
+func updateQuery(query string, mapping []SourceMapping) (string, []string, error) {
```

Comment:
Can this and the replaceSourceName be the same function? Seems like very similar code